cmd/rook/ceph/osd.go

```diff
@@ -118,16 +133,52 @@ func runFilestoreDeviceOSD(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-// Provision a device or directory for an OSD
-func prepareOSD(cmd *cobra.Command, args []string) error {
+func verifyConfigFlags(configCmd *cobra.Command) error {
 	required := []string{"cluster-id", "node-name"}
-	if err := flags.VerifyRequiredFlags(provisionCmd, required); err != nil {
+	if err := flags.VerifyRequiredFlags(configCmd, required); err != nil {
 		return err
 	}
 	required = []string{"cluster-name", "mon-endpoints", "mon-secret", "admin-secret"}
 	if err := flags.VerifyRequiredFlags(osdCmd, required); err != nil {
 		return err
 	}
+	return nil
+}
+
+func writeOSDConfig(cmd *cobra.Command, args []string) error {
+	if err := verifyConfigFlags(osdConfigCmd); err != nil {
+		return err
+	}
+	if osdID == -1 {
+		return fmt.Errorf("osd id not specified")
+	}
+
+	clientset, _, _, err := rook.GetClientset()
+	if err != nil {
+		rook.TerminateFatal(fmt.Errorf("failed to init k8s client. %+v\n", err))
+	}
+
+	context := createContext()
+	context.Clientset = clientset
+	commonOSDInit(osdConfigCmd)
+	locArgs, err := client.FormatLocation(cfg.location, cfg.nodeName)
+	if err != nil {
+		rook.TerminateFatal(fmt.Errorf("invalid location. %+v\n", err))
+	}
+	crushLocation := strings.Join(locArgs, " ")
+	kv := k8sutil.NewConfigMapKVStore(clusterInfo.Name, clientset, metav1.OwnerReference{})
```

Comment:
Do you think it would be in general an idea to remove the "kvstore" stuff as Kubernetes is our only "store" and talking "directly" to the ConfigMap may make things easier? Not in this PR, but just as a thought.